src/main/kotlin/com/exactpro/th2/common/message/BuilderExtension.kt

```diff
@@ -0,0 +1,164 @@
+val ParsedMessageBuilderImpl.body: BodyBuilder
+    get() = BodyBuilder(this)
+
+val RawMessageBuilder<*>.metadata: RawMetadataBuilder
+    get() = metadataBuilder()
+
+operator fun RawMetadataBuilder.invoke(block: RawMetadataBuilder.() -> Unit) = block()
+
+fun MessageFactory.createParsedMessage(block: ParsedMessageBuilderImpl.() -> Unit): Message = createParsedMessage()(block)
+
+fun MessageFactory.createRawMessage(block: RawMessageBuilderImpl.() -> Unit): RawMessage = createRawMessage()(block)
+
+fun main() {
+    val commonFactory = CommonFactory.createFromArguments("-c", "src/test/resources/test_load_dictionaries")
+    val messageFactory = commonFactory.messageFactory
+    val bookName = commonFactory.boxConfiguration.bookName!!
+    testParsedMessage(messageFactory, bookName)
+    testRawMessage(messageFactory, bookName)
+    testEvent(commonFactory.eventFactory, bookName)
+}
+
+fun testParsedMessage(factory: MessageFactory, bookName: String) {
+    val message = factory.createParsedMessage {
+        setParentEventId("eventId", bookName)
+        metadata {
+            setSessionAlias("test")
+            setDirection(Direction.SECOND)
+            setSequence(1)
+            addSubsequence(2)
+            addSubsequence(3)
+            setBookName(bookName)
+            setTimestamp(Instant.now())
+            setMessageType("type")
+            putProperty("propertyKey", "propertyValue")
+            setProtocol("protocol")
+        }
+        body {
+            "A" to 5
+            "B" to listOf(1, 2, 3)
+            "C" toMessage {
+                "A" to 5
+                "B" to listOf(1, 2, 3)
+            }
+            "D" += {
+                "A" to 42
+            }
+            "E" toMessages listOf<BodyBuilder.() -> Unit>(
+                {
+                    "A" to 4
+                },
+                {
+                    "A" to 5
+                }
+            )
+        }
+    }
+    println(message.toJson(false))
+}
+
+fun testRawMessage(factory: MessageFactory, bookName: String) {
+    val message = factory.createRawMessage {
+        setParentEventId("eventId", bookName)
+        setBody("body".toByteArray())
+        metadata {
+            setSessionAlias("test")
+            setDirection(Direction.SECOND)
+            setSequence(1)
+            addSubsequence(2)
+            addSubsequence(3)
+            setBookName(bookName)
+            setTimestamp(Instant.now())
+            putProperty("propertyKey", "propertyValue")
+            setProtocol("protocol")
+        }
+    }
+    println(message.toJson(false))
+}
+
+fun testEvent(eventFactory: IEventFactory, bookName: String) {
+    val event = Event.start(eventFactory)
+        .bookName(bookName)
+        .status(Event.Status.PASSED)
+        .name("name")
+        .type("type")
+        .bodyData(EventUtils.createMessageBean("bodyData"))
+        .messageID(MessageID.newBuilder().build())
+        .toProto(null)
+    println(event.toJson(false))
+}
```

Comment:
We should remove test methods

Reply:
Of course. It's just an example and it will be deleted.